state-chain/pallets/cf-broadcast/src/lib.rs

```diff
@@ -930,29 +792,23 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 
 			Timeouts::<T, I>::append(
 				frame_system::Pallet::<T>::block_number() + T::BroadcastTimeout::get(),
-				broadcast_attempt.broadcast_attempt_id,
+				broadcast_id,
 			);
 
 			Self::deposit_event(Event::<T, I>::TransactionBroadcastRequest {
-				broadcast_attempt_id: broadcast_attempt.broadcast_attempt_id,
+				broadcast_id,
 				nominee: nominated_signer,
 				transaction_payload: broadcast_attempt.transaction_payload,
 				transaction_out_id: broadcast_attempt.transaction_out_id,
 			});
 		} else {
-			log::warn!(
-				"Failed to select a signer for broadcast {:?}. Scheduling Retry",
-				broadcast_attempt.broadcast_attempt_id
-			);
-			Self::schedule_for_retry(&broadcast_attempt);
+			Self::abort_broadcast(broadcast_id);
```

Comment:
This is quite different to what was mentioned in the issue. The reason we want to check the length of failed broadcasters to decide if we should abort or not and *not* whether we can nominate is because people can be excluded for reasons *other* than just the fact they failed to broadcast this particular transaction. If you look inside `nominate_broadcaster` nodes can be excluded for other reasons:
```rust
let mut all_excludes = Reputation::validators_suspended_for(&[
			Offence::FailedToBroadcastTransaction,
			Offence::MissedHeartbeat,
		]);
```

These are temporary suspensions. After some time these suspensions will wear off and then this nomination can succeed - so there's a bit of an issue here.

I think using the approach as the code was before, where we abort when the failure actually occurs is a bit better. Doing it this way also means we don't do unnecessary extra things before aborting like this:

```rust
		T::TransactionBuilder::refresh_unsigned_data(&mut broadcast_attempt.transaction_payload);
		TransactionMetadata::<T, I>::insert(
			broadcast_id,
			<<T::TargetChain as Chain>::TransactionMetadata>::extract_metadata(
				&broadcast_attempt.transaction_payload,
			),
		);		
```

Reply:
Acknowledged. I've put the abort checking logic into "handle_broadcast_failure".
Note with this, if a broadcast has not reported failure, cannot be nominated (due to failed heartbeat for example) and cannot recover, then we will keep retrying forever (atleast until the next epoch?)

Reply:
Yep this is how it should behave. It won't retry forever, but until a suspension has elapsed for one of the authorities.